doc/default.nix

```diff
@@ -16,40 +13,75 @@ let
   # Prefix to remove from option declaration file paths.
   rootPrefix = toString ../. + "/";
 
-  nixosConfiguration = nixosSystem {
-    inherit system;
-    modules = [
-      inputs.home-manager.nixosModules.home-manager
-      inputs.self.nixosModules.stylix
-      ./settings.nix
-    ];
+  # A stub pkgs used while evaluating the stylix modules for the docs
+  noPkgs = {
+    # Needed for type-checking
+    inherit (pkgs) _type;
+
+    # Permit access to (pkgs.formats.foo { }).type
+    formats = builtins.mapAttrs (_: fmt: args: {
+      inherit (fmt args) type;
+    }) pkgs.formats;
   };
 
-  homeConfiguration = homeManagerConfiguration {
-    inherit pkgs;
-    modules = [
-      inputs.self.homeModules.stylix
-      ./settings.nix
-      {
-        home = {
-          homeDirectory = "/home/book";
-          stateVersion = "22.11";
-          username = "book";
+  # A stub config used while evaluating the stylix modules for the docs
+  #
+  # To resolve #98, this should be simplified to `noConfig = {}`.
+  # However, currently we access option-declaring functions via
+  # `config.lib.stylix`.
+  noConfig =
+    let
+      configuration = evalDocs {
+        # The config.lib option, as found in NixOS and home-manager.
+        # Currently required by the `target.nix` module.
+        options.lib = lib.mkOption {
+          type = lib.types.attrsOf lib.types.attrs;
+          description = ''
+            This option allows modules to define helper functions, constants, etc.
+          '';
+          default = { };
+          visible = false;
         };
-      }
-    ];
-  };
+
+        # The target.nix module defines functions that are currently needed to
+        # declare options
+        imports = [ ../stylix/target.nix ];
+      };
+    in
+    {
+      lib.stylix = {
+        inherit (configuration.config.lib.stylix)
+          mkEnableIf
+          mkEnableTarget
+          mkEnableTargetWith
+          mkEnableWallpaper
+          ;
+      };
+    };
+
+  evalDocs =
+    module:
+    lib.evalModules {
+      specialArgs.config = noConfig;
+      modules = lib.toList module ++ [
+        ./eval_compat.nix
+        { _module.args.pkgs = noPkgs; }
+      ];
```

Comment:
For reference, the NixOS manual implements their restrictions here: https://github.com/NixOS/nixpkgs/blob/d6a5810feb1772e665138ba97ecf1ff033e7feb6/nixos/modules/misc/documentation.nix#L75-L106

(I wrote these before looking at the NixOS impl, they take a very different approach)

Reply:
Note I'm not linking to the NixOS impl because I think Stylix should copy it. It is different because it has different goals.

Rather I'm linking to it for reference (including my own future reference!)

IMO the current impl added by this PR is simple enough to work well for stylix, while being effective at keeping accidentally dynamic evaluation out of the docs.